arch/arm/core/cortex_m/prep_c.c

```diff
@@ -24,7 +24,7 @@
 #include <arch/arm/cortex_m/cmsis.h>
 #include <string.h>
 
-#ifdef CONFIG_ARMV6_M
+#if defined(CONFIG_ARMV6_M) || defined(CONFIG_ARMV8_M_BASELINE)
```

Comment:
Maybe we could define some `ARMV6_V8_BASELINE` to avoid having to `||` all the time?

Reply:
@carlescufi  I do not have a strong opinion on that. Having the compile-time _OR_ directive is more descriptive and educative; a composite define saves space, and reduces risk of forgetting a directive.

Reply:
I'm with @carlescufi - we should have a single option here, its less error prone.

How about CONFIG_ARMV6_AND_V8_BASELINE

Reply:
@carlescufi @galak  As a single Kconfig option, I would propose:

CONFIG_ARMV6_M_ARMV8_M_BASELINE (we should keep the _M_ in the name)

And, also, in Kconfig, I would not define this option directly, but as an option dependent on the "parent" options, as follows:


config ARMV6_M_ARMV8_M_BASELINE
    bool
    depends on ARMV6_M or ARMV8_M_BASELINE
    default n
    .........

Does this look good enough?

Reply:
What do you mean by not define this option directly?

I think we should just 'select' it in the right place. depends on gets a little messy here.

Reply:
@galak I meant the following:
ARMV6_M, ARMV8_M_BASELINE would be "selected" by the corresponding Cortex cores (for example, CM0 and CM33, respectively). Then ARMV6_M_ARMV8_M_BASELINE would be "in"directly selected via the Kconfig option definition I presented above.

But perhaps, it is an overkill, and we could just select the common option in the cortex cores.

Reply:
@galak @carlescufi  I updated the PR based on your suggestion to use a composite Kconfig option.